com.sap.cloud.lm.sl.cf.core/src/main/java/com/sap/cloud/lm/sl/cf/core/parser/UriParametersParser.java

```diff
@@ -9,35 +9,42 @@
 import java.util.List;
 import java.util.Map;
 import java.util.Set;
+import java.util.stream.Collectors;
 
 import com.sap.cloud.lm.sl.cf.core.model.SupportedParameters;
 
 public class UriParametersParser implements ParametersParser<List<String>> {
 
     private boolean portBasedRouting;
+    private boolean includeProtocol;
     private String defaultHost;
     private String defaultDomain;
     private Integer defaultPort;
     private String hostParameterName;
     private String domainParameterName;
     private String portParameterName;
     private String routePath;
+    private String protocol;
 
-    public UriParametersParser(boolean portBasedRouting, String defaultHost, String defaultDomain, Integer defaultPort, String routePath) {
+    public UriParametersParser(boolean portBasedRouting, String defaultHost, String defaultDomain, Integer defaultPort, String routePath,
+        boolean includeProtocol, String protocol) {
         this(portBasedRouting, defaultHost, defaultDomain, defaultPort, SupportedParameters.HOST, SupportedParameters.DOMAIN,
-            SupportedParameters.PORT, routePath);
+            SupportedParameters.PORT, routePath, includeProtocol, protocol);
     }
 
     public UriParametersParser(boolean portBasedRouting, String defaultHost, String defaultDomain, Integer defaultPort,
-        String hostParameterName, String domainParameterName, String portParameterName, String routePath) {
+        String hostParameterName, String domainParameterName, String portParameterName, String routePath, boolean includeProtocol,
```

Comment:
Can we have only one additional parameter "protocol" and use it when not null.
This is the case with "routePath"

Reply:
The protocol is not null even on CF systems as it still gets calculated to HTTP.

Reply:
OK